digits/frameworks/torch_framework.py

```diff
@@ -0,0 +1,162 @@
+# Copyright (c) 2015, NVIDIA CORPORATION.  All rights reserved.
+
+import os
+import digits
+import re
+import subprocess
+import time
+import tempfile
+import flask
+
+from framework import Framework
+from digits import utils
+from digits.config import config_value
+from digits.model.tasks import TorchTrainTask
+from digits.utils import subclass, override
+from errors import Error, NetworkVisualizationError, BadNetworkError
+
+@subclass
+class TorchFramework(Framework):
+
+    """
+    Defines required methods to interact with the Torch framework
+    """
+
+    # short descriptive name
+    NAME = 'Torch (experimental)'
```

Comment:
The 'descriptive' name is only used in the navigation pane [there](https://github.com/gheinrich/DIGITS/blob/4e2a55ddb1d03777870199a813c117a346128113/digits/templates/models/images/classification/new.html#L349) on the model creation pane. The model job keeps a link to the framework through the train task and its `framework_id` field [there](https://github.com/gheinrich/DIGITS/blob/4e2a55ddb1d03777870199a813c117a346128113/digits/model/tasks/train.py#L55) - the framework ID is what's shown on the home page and in the previous networks tab. We can change the descriptive name and that will change the display on the navigation pane however the framework ID will still be 'torch'. When we want to support Torch9 we can create a new framework ID to denote the change in major version. Is that OK?